src/algebra/char_p/basic.lean

```diff
@@ -473,6 +474,10 @@ begin
   rwa ← hc
 end
 
+/-- The characteristic of a finite ring cannot be zero. -/
+lemma ring_char.char_ne_zero_of_fintype (R) [non_assoc_ring R] [fintype R] : ring_char R ≠ 0 :=
```

Comment:
I think this would be a better name:
```suggestion
lemma ring_char_ne_zero_of_fintype [fintype R] : ring_char R ≠ 0 :=
```

Reply:
OK. I changed `[comm_ring R]` on line 446 to `[non_assoc_ring R]`, but then had to use `(R) [ring R] [fintype R]` for `char_p_of_prime_pow_injective` (`non_assoc_ring` is not enough for the statement to type check).
One can probably weaken the assumptions in a lot of these lemmas, but I don't have the time to do that systematically.